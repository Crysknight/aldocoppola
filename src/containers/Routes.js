import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Switch, Route } from 'react-router';
import { Link } from 'react-router-dom';

import * as actions from '../actions';

// Import the containers related to paths
import New from './New';
import NewToo from './NewToo';

class Routes extends Component {

  // Here we assign, which container will render at which path
  // We do it here to make it possible to map the paths' reducer to an array of Route components
  componentWillMount() {
    let paths = this.props.paths;
    paths.app.component = New;
    paths.newToo.component = NewToo;
  }

  render() {
    // Here we create an array of paths in order to map through it
    let paths = this.props.paths;
    let pathArray = [];
    for (let path in paths) {
      pathArray.push(paths[path]);
    }
    console.log(paths.getAppSwitch(this.props.location.pathname));
    return (
      <div id="ac_layout">
        <Link 
          className="app-switch" 
          to={`${paths.getAppSwitch(this.props.location.pathname)}`} 
        >+</Link>
        <Switch>
          {pathArray.map((path, index) => {
            return (
              <Route 
                key={index}
                exact
                path={`${paths.getPath(this.props.location.pathname, path)}`} 
                component={path.component} 
              />
            );
          })}
        </Switch>
      </div>
    );
  }

}

function mapStateToProps(state) {
  return {
    paths: state.paths
  };
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators({
    
  }, dispatch);
}

export default connect(mapStateToProps, matchDispatchToProps)(Routes);