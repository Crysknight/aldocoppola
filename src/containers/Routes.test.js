import React from 'react';
import Routes from './Routes';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import allReducers from '../reducers';

const locationRoot = {
	pathname: '/_appliance/'
};

const locationNested = {
	pathname: '/test/bauble.html/_appliance/'
};

const locationChild = {
	pathname: '/_appliance/new-too/'
};

const locationUnrelated = {
	pathname: '/test/bauble.html'
}

describe('Routes', () => {
	it('matches the snapshot on the rootpath', () => {
		const store = createStore(allReducers);
		const component = renderer.create(
			<Provider store={store}>
				<MemoryRouter initialEntries={[ '/_appliance/' ]}>
					<Routes location={locationRoot} />
				</MemoryRouter>
			</Provider>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('matches the snapshot on the nested path', () => {
		const store = createStore(allReducers);
		const component = renderer.create(
			<Provider store={store}>
				<MemoryRouter initialEntries={[ '/test/bauble.html/_appliance/' ]}>
					<Routes location={locationNested} />
				</MemoryRouter>
			</Provider>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('child path of the app matches the snapshot on the rootpath', () => {
		const store = createStore(allReducers);
		const component = renderer.create(
			<Provider store={store}>
				<MemoryRouter initialEntries={[ '/_appliance/new-too/' ]}>
					<Routes location={locationChild} />
				</MemoryRouter>
			</Provider>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('doesn\'t render the app everywhere', () => {
		const store = createStore(allReducers);
		const component = renderer.create(
			<Provider store={store}>
				<MemoryRouter initialEntries={[ '/test/bauble.html' ]}>
					<Routes location={locationUnrelated} />
				</MemoryRouter>
			</Provider>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

});