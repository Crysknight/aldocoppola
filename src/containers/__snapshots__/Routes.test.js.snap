// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes child path of the app matches the snapshot on the rootpath 1`] = `
<div
  id="ac_layout"
>
  <a
    className="app-switch"
    href="/"
    onClick={[Function]}
  >
    +
  </a>
  <div
    className="new-too"
  >
    <p>
      Hello, it's NewToo
    </p>
    <a
      href="/_appliance/"
      onClick={[Function]}
    >
      Go to new
    </a>
  </div>
</div>
`;

exports[`Routes doesn't render the app everywhere 1`] = `
<div
  id="ac_layout"
>
  <a
    className="app-switch"
    href="/test/bauble.html/_appliance/"
    onClick={[Function]}
  >
    +
  </a>
</div>
`;

exports[`Routes matches the snapshot on the nested path 1`] = `
<div
  id="ac_layout"
>
  <a
    className="app-switch"
    href="/test/bauble.html"
    onClick={[Function]}
  >
    +
  </a>
  <div
    className="new"
  >
    <div
      className="ac-header"
    >
      <h3>
        Выбрать центр красоты
      </h3>
    </div>
    <p>
      Hello, it's new
    </p>
    <a
      href="/test/bauble.html/_appliance/new-too/"
      onClick={[Function]}
    >
      Go to new-too
    </a>
  </div>
</div>
`;

exports[`Routes matches the snapshot on the rootpath 1`] = `
<div
  id="ac_layout"
>
  <a
    className="app-switch"
    href="/"
    onClick={[Function]}
  >
    +
  </a>
  <div
    className="new"
  >
    <div
      className="ac-header"
    >
      <h3>
        Выбрать центр красоты
      </h3>
    </div>
    <p>
      Hello, it's new
    </p>
    <a
      href="/_appliance/new-too/"
      onClick={[Function]}
    >
      Go to new-too
    </a>
  </div>
</div>
`;
